extensions/jira/deployment/src/main/java/org/apache/camel/quarkus/component/jira/deployment/JiraProcessor.java

```diff
@@ -55,8 +54,121 @@ NativeImageResourceBuildItem nativeImageResources() {
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Permission.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Permissions.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Priority.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Project.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.ProjectRole.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Resolution.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.RoleActor.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.SearchResult.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.SecurityLevel.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.ServerInfo.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Session.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.SessionCookie.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.StandardOperation.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Status.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Subtask.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.TimeTracking.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Transition.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.User.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Version.class));
+        items.add(new ReflectiveClassBuildItem(true, false,
+                com.atlassian.jira.rest.client.api.domain.VersionRelatedIssuesCount.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Visibility.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Votes.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Watchers.class));
+        items.add(new ReflectiveClassBuildItem(true, false, com.atlassian.jira.rest.client.api.domain.Worklog.class));
```

Comment:
It seems that we have to add all classes in `com.atlassian.jira.rest.client.api.domain` package?

Reply:
I am curious as to why this has not been a requirement before. I am guessing it's because you (or the Kamelet) is doing JSON serialization on these types?

Reply:
Given the example route to retrieve jira issues, the message object is [`Issue`](https://bitbucket.org/atlassian/jira-rest-java-client/src/master/api/src/main/java/com/atlassian/jira/rest/client/api/domain/Issue.java) class, which has a hard dependency on many other jira classes. And it would be imprecise to say which classes are really required, as the issue itself could contain any of the attributes. 
I will try to reduce the number of classes.

Reply:
Maybe we can try to catch all those classe in this package, instead of naming each of them?